File: 7. Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        res = 0
        i = 0
        is_negative = True if x < 0 else False
        x = abs(x)
        while x > 0:
            rem = x % 10
            res = res * 10 + rem
            x = x // 10
            i += 1

        res = -res if is_negative else res
        if -2 ** 31 >= res or res >= 2 ** 31 - 1:
            return 0
        return res

File: javascript/7. Reverse Integer.js
var reverse = function(x) {
    const isNegative = x < 0;
    const xStrArr = Math.abs(x).toString().split("");
    const reversStr = xStrArr.reverse().join("");
    const num = Number(reversStr);
    if (isNegative && num > Math.pow(2, 31)) {
        return 0;
    }
    if (!isNegative && num > Math.pow(2, 31) - 1) {
        return 0;
    }
    return isNegative ? -num : num;
};
